server/src/infra/mail.rs

```diff
@@ -17,6 +17,7 @@ async fn send_email(to: Mailbox, subject: &str, body: String, options: &MailOpti
         &to, &from, &options.user, &options.server, options.port
     );
     let email = Message::builder()
+        .message_id(Some(format!("<{}@{}>", crate::util::get_uuid(), smtp_from.split('@').collect::<Vec<&str>>()[1])))
```

Comment:
I don't think we have a `crate::util::get_uuid()` function, and there is no local variable `smtp_from`.

Could you try compiling and testing locally?

Reply:
**how to run and tet locally? **

Reply:
You can check the section on building from source in the readme: https://github.com/lldap/lldap#from-source

Once you have it built, you can run tests with `cargo test`.

Reply:
@nitnelave why am i getting split() method not available? it is an inbuilt method.help me to resolve it.

Reply:
It's a built-in method for which type? And what type are you calling it on?

Reply:
I am calling it like thet from.split('@').

Reply:
Yes, I can see that. What is the type of `from` ?

Reply:
I think I have to convert it to string first then call the split(). Right?

Reply:
That would be one way. But try answering the question first: what is the type of `from`?

Reply:
I think message type.

Reply:
Nope. To make sure that you know which type it is, please send me a link to the definition of the type of `from`, or the docs for its type.

Reply:
Option<Mailbox>

Reply:
That would be the type of `options.from`. The type of `from` is just `Mailbox`. Now, using the docs, can you find a different way to extract the domain from the address?

Reply:
Allow me some time i will let you know.